### PR TITLE
Add support for repacking .zip files

### PR DESCRIPTION
**Pull request by JohnTortugo**
Closes: #916 

Currently zip files aren't being repacked because the logic assumes that all files that need to be repacked also need to be signed. This PR adds an option to sign .zip files - which aren't signed themselves.

### CODE REVIEW

**Comment by tmat** [comment]
>             if (fileSignInfo.SignInfo.ShouldSign)

I think we should add containers that are not signed themselves to _filesToSign and then skip the actual signing if the sign info says so.
So this condition should be ShouldSign || IsZipContainer.

---
Refers to: src/Microsoft.DotNet.SignTool/src/Configuration.cs:140 in a12393f. [](commit_id = a12393fba57aa0467af5983066aff30563ac61fc, deletion_comment = False)

**Comment by tmat** [comment]
A test need to be added with ItemsToSign = [zip file, package that contains a zip file, zip file that contains another zip file].

**Comment by JohnTortugo** [comment]
Done. PTAL:

- Added new test cases and
- Refactored following your suggestion

**Comment by JohnTortugo** [comment]
@tmat can you take a look?

File: src/Microsoft.DotNet.SignTool/src/BatchSignUtil.cs [hunk 1]
```diff
@@ -91,9 +91,9 @@ private bool SignFiles()
 
             bool signFiles(IEnumerable<FileSignInfo> files)
             {
-                var filesToSign = files.ToArray();
+                var filesToSign = files.Where(fileInfo => fileInfo.SignInfo.ShouldSign).ToArray();
 
-                _log.LogMessage(MessageImportance.High, $"Signing Round {round}: {filesToSign.Length} files to sign.");
+                _log.LogMessage(MessageImportance.High, $"Signing Round {round}: {filesToSign.Length} files to sign (non-signable containers skipped).");
```
Comment (tmat) [inline src/Microsoft.DotNet.SignTool/src/BatchSignUtil.cs]: >(non-signable containers skipped) [](start = 116, length = 33)

I don't think this is necessary.

File: src/Microsoft.DotNet.SignTool.Tests/SignToolTests.cs [hunk 2]
```diff
@@ -458,6 +459,8 @@ public void ZipFile()
   <Authenticode>Microsoft400</Authenticode>
 </FilesToSign>
 ",
+
+$@"" // for the .zip file that isn't actually signed
```
Comment (tmat) [inline src/Microsoft.DotNet.SignTool.Tests/SignToolTests.cs]: This is odd. We shouldn't create signing request with no files.

File: src/Microsoft.DotNet.SignTool/src/BatchSignUtil.cs (hunk 1 above)
Comment (tmat) [inline src/Microsoft.DotNet.SignTool/src/BatchSignUtil.cs]: Return here if `filesToSign.Length == 0`

Comment (tmat) [inline src/Microsoft.DotNet.SignTool/src/BatchSignUtil.cs]: >_ [](start = 16, length = 1)

Return `true` here if `filesToSign.Length == 0`

**Pull request merged**